lua/lsp/null-ls/init.lua

```diff
@@ -1,44 +1,26 @@
 local M = {}
 
-function M.list_supported_provider_names(filetype)
-  local names = {}
-
-  local formatters = require "lsp.null-ls.formatters"
-  local linters = require "lsp.null-ls.linters"
-
-  vim.list_extend(names, formatters.list_supported_names(filetype))
-  vim.list_extend(names, linters.list_supported_names(filetype))
-
-  return names
-end
-
-function M.list_unsupported_provider_names(filetype)
-  local names = {}
-
-  local formatters = require "lsp.null-ls.formatters"
-  local linters = require "lsp.null-ls.linters"
-
-  vim.list_extend(names, formatters.list_unsupported_names(filetype))
-  vim.list_extend(names, linters.list_unsupported_names(filetype))
-
-  return names
-end
-
--- TODO: for linters and formatters with spaces and '-' replace with '_'
-function M.setup(filetype, options)
-  options = options or {}
-
-  local ok, _ = pcall(require, "null-ls")
-  if not ok then
-    require("core.log"):error "Missing null-ls dependency"
+local Log = require "core.log"
+local formatters = require "lsp.null-ls.formatters"
+local linters = require "lsp.null-ls.linters"
+
+function M:setup()
+  local status_ok, null_ls = pcall(require, "null-ls")
+  if not status_ok then
+    Log:error "Missing null-ls dependency"
     return
   end
 
-  local formatters = require "lsp.null-ls.formatters"
-  local linters = require "lsp.null-ls.linters"
-
-  formatters.setup(filetype, options)
-  linters.setup(filetype, options)
+  null_ls.config()
+  require("lspconfig")["null-ls"].setup {}
+  for _, filetype in pairs(lvim.lang) do
+    if filetype.formatters then
```

Comment:
I think that once it is autogenerated (which you have already partially done) we can remove these conditions